vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,381 @@
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_scope": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A list of IP scopes for the network",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"is_inherited": &schema.Schema{
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "True if the IP scope is inherit from parent network",
+						},
+						"gateway": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Gateway of the network",
+							ValidateFunc: singleIP(),
+						},
+						"netmask": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Network mask",
+							ValidateFunc: singleIP(),
+						},
+						"dns1": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Primary DNS server",
+							ValidateFunc: singleIP(),
+						},
+						"dns2": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Secondary DNS server",
+							ValidateFunc: singleIP(),
+						},
+						"dns_suffix": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Dns suffix",
+						},
+						"ip_range": &schema.Schema{
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "IP ranges used for static pool allocation in the network",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"start": &schema.Schema{
+										Type:         schema.TypeString,
+										Required:     true,
+										Description:  "Start address of the IP range",
+										ValidateFunc: singleIP(),
+									},
+									"end": &schema.Schema{
+										Type:         schema.TypeString,
+										Required:     true,
+										Description:  "End address of the IP range",
+										ValidateFunc: singleIP(),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"vsphere_networks": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A list of port groups that back this network. Each referenced DV_PORTGROUP or NETWORK must exist on a vCenter server registered with the system.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"vcenter": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The vCenter server name",
+						},
+						"vsphere_network": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The name of the port group",
```

Comment:
Maybe better "The name of a **vSphere network or** a port group"?

Reply:
I am not sure it's right. I think should be one of them.

Reply:
OK, port groups they are then.